powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/v3/PowerAuthServiceImpl.java

```diff
@@ -205,7 +205,34 @@ public PrepareActivationResponse prepareActivation(PrepareActivationRequest requ
     @Override
     @Transactional
     public CreateActivationResponse createActivation(CreateActivationRequest request) throws GenericServiceException {
-        throw new IllegalStateException("Not implemented yet.");
+        try {
+            // Get request parameters
+            String userId = request.getUserId();
+            Date activationExpireTimestamp = XMLGregorianCalendarConverter.convertTo(request.getTimestampActivationExpire());
+            Long maxFailedCount = request.getMaxFailureCount();
+            String applicationKey = request.getApplicationKey();
+            byte[] ephemeralPublicKey = BaseEncoding.base64().decode(request.getEphemeralPublicKey());
+            byte[] mac = BaseEncoding.base64().decode(request.getMac());
```

Comment:
What if the request values are `null`?

Reply:
Good catch, I will fix it in all occurrences.

Reply:
I will add new error `INVALID_REQUEST` and add null checks to all methods in `PowerAuthServiceImpl`, such validation is generally missing.

Reply:
I added nullability checks into all service methods, added the error as well as fixed missing error messages for other errors.